src/movies.js

```diff
@@ -1,15 +1,53 @@
+
+
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+    return movies.map(function(movie){
+        return movie.director
+      })
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies) {
+    const spielbergDrama = movies.filter(movie => movie.director === "Steven Spielberg" && movie.genre.includes("Drama"))
+    return spielbergDrama.length
+  }
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+function ratesAverage(movies) {
+    if (movies.length === 0) {return 0};
+    let totalRates = movies.reduce(function(sum, movie) {
```

Comment:
to solve the error
`Return average even if one of the movies does not have rate!`
you can just add an if statement for the movies that hasn't a rate
```if (!movie.rate) return sum;```
so for the ones that hasn't a rate value you can just return the sum.